prepare/metrics/string_containment.py

```diff
@@ -0,0 +1,33 @@
+from src.unitxt import add_to_catalog
+from src.unitxt.metrics import StringContainment
+from src.unitxt.test_utils.metrics import test_metric
+
+metric = StringContainment()
+
+predictions = ["barak obama is a politician", "David Gilmour is an English guitarist"]
+references = [["politician | politic | pol"], ["artist"]]
```

Comment:
```suggestion
references = [["politician", "politic, "pol"], ["artist"]]
```

Reply:
I think it will be better to use the option of providing multiple references to the metric. Then we are not limited to this specific encoding with the '|' separator, and the metric is more general.
This though requires some other changes, pls see my other comments.

Reply:
done, please see my comment here https://github.com/IBM/unitxt/pull/394#discussion_r1429117281